src/wp-includes/class-wp-db-partial-query.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * WordPress partial query builder class.
+ *
+ * @package WordPress
+ * @subpackage Database
+ */
+/**
+ * Class WP_DB_Partial_Query.
+ *
+ * This class is used as a query builder to generate partial escaped queries.
+ */
+final class WP_DB_Partial_Query {
+
+	/**
+	 * Actual prepared query.
+	 *
+	 * @var string|void
+	 */
+	private $query;
+
+	/**
+	 * WP_DB_Partial_Query constructor.
+	 *
+	 * Accepts exapctly the same parameters as $wpdb->prepare.
+	 *
+	 * @param $query
+	 * @param ...$args
+	 */
+	public function __construct( $query, ...$args ) {
+		global $wpdb;
+		$this->query = $wpdb->prepare( $query, ...$args );
+	}
+
+	/**
+	 * Get prepared query.
+	 *
+	 * @return string|void
+	 */
+	public function get_query() {
+		return $this->query;
+	}
+
+
+	/**
+	 * Implementing this magic method allows object of this class to be interpolated in sprintf, vprintf etc methods.
+	 */
```

Comment:
```suggestion
	/**
	 * Implementing this magic method allows object of this class to be interpolated in sprintf, vprintf etc methods.
	 *
	 * @return string The interpolated WP_DB_Partial_Query object.
	 */
```

For docs consistency, even `__toString()` should have a `@return` annotation. [See here](https://github.com/WordPress/wordpress-develop/blob/954e9c153fe9d1b4f74d76bab9dd5175fd36cebc/src/wp-includes/class-wp-theme.php#L473) for another example in core.